risc0/cargo-risczero/src/commands/datasheet/mod.rs

```diff
@@ -0,0 +1,487 @@
+use clap::ValueEnum;
+use risc0_zkp::MAX_CYCLES_PO2;
+use risc0_zkvm::{
+    ApiClient, Asset, AssetRequest, ExecutorEnv, ExecutorEnvBuilder, ProveInfo, ProverOpts,
+    ReceiptClaim, SegmentInfo, SessionInfo, SuccinctReceipt, VerifierContext, RECURSION_PO2,
+};
+use serde_with::{serde_as, DurationSeconds};
+use tabled::{settings::Style, Table, Tabled};
+
+/// `cargo risczero datasheet`
+#[derive(clap::Parser)]
+#[non_exhaustive]
+pub struct Datasheet {
+    /// Filter which benchmarks to run.
+    #[arg(name = "BENCHMARKS")]
+    pub filter: Vec<Benchmark>,
+
+    /// Path to the RISC Zero server.
+    #[arg(long, value_name = "PATH", env = "RISC0_SERVER_PATH")]
+    pub r0vm: Option<PathBuf>,
+
+    /// Path to the JSON results file.
+    #[arg(long, value_name = "PATH")]
+    pub json: Option<PathBuf>,
+
+    /// Max size for composite runs.
+    #[arg(long, default_value_t = 20, value_parser = parse_po2)]
+    pub max_po2: u32,
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, ValueEnum)]
+#[non_exhaustive]
+pub enum Benchmark {
+    Execute,
+    Composite,
+    Lift,
+    Join,
+    Succinct,
+    Identity,
+}
+
+/// Pre-compiled program that simply loops `count: u32` times (read from stdin).
+const LOOP_ELF: &[u8] = include_bytes!("loop.bin");
+
+/// Powers-of-two for cycles paired with the number of loop iterations used to
+/// achieve that many cycles.
+const CYCLES_PO2_ITERS: &[(u32, u32)] = &[
+    (15, 1),               // 15, 32K
+    (16, 1024 * 8),        // 16, 64K
+    (17, 1024 * 32),       // 17, 128K
+    (18, 1024 * 64),       // 18, 256K
+    (19, 1024 * 128),      // 19, 512K
+    (20, 1024 * 256),      // 20, 1M
+    (21, 1024 * 256 * 3),  // 21, 2M
+    (22, 1024 * 256 * 7),  // 22, 4M
+    (23, 1024 * 256 * 15), // 23, 8M
+    (24, 1024 * 256 * 31), // 24, 16M
+];
+
+const MIN_CYCLES_PO2: u32 = CYCLES_PO2_ITERS[0].0;
+
+fn parse_po2(s: &str) -> Result<u32, String> {
+    let po2 = s.parse().map_err(|_| format!("`{s}` must be an integer"))?;
+
+    if (MIN_CYCLES_PO2..=MAX_CYCLES_PO2 as u32).contains(&po2) {
+        Ok(po2)
+    } else {
+        Err(format!(
+            "po2 must be in range: {MIN_CYCLES_PO2}-{MAX_CYCLES_PO2}"
+        ))
+    }
+}
+
+impl Datasheet {
+    /// Returns which benchmarks the user has chosen to run.
+    ///
+    /// Each benchmark is provided at most once in enum case order.
+    fn benchmarks(&self) -> impl Iterator<Item = Benchmark> + '_ {
+        Benchmark::value_variants()
+            .iter()
+            .filter(|benchmark| self.filter.is_empty() || self.filter.contains(benchmark))
+            .copied()
+    }
+
+    /// Creates a connection to the zkVM server (`r0vm` executable).
+    ///
+    /// The path is provided by `--r0vm`, `$RISC0_SERVER_PATH`, or searching
+    /// `$PATH`.
+    fn connect_to_server(&self) -> Result<ApiClient> {
+        let server_path = self.r0vm.as_deref().unwrap_or("r0vm".as_ref());
+        ApiClient::new_sub_process(server_path)
```

Comment:
```suggestion
        ApiClient::from_env()
```

Reply:
Do we not want this configurable with a CLI flag?